kedro/framework/cli/starters.py

```diff
@@ -223,19 +264,16 @@ def new(  # noqa: PLR0913
     **kwargs,
 ):
     """Create a new kedro project."""
-    if checkout and not starter_alias:
-        raise KedroCliError("Cannot use the --checkout flag without a --starter value.")
-
-    if directory and not starter_alias:
-        raise KedroCliError(
-            "Cannot use the --directory flag without a --starter value."
-        )
-
-    if (selected_tools or example_pipeline) and starter_alias:
-        raise KedroCliError(
-            "Cannot use the --starter flag with the --example and/or --tools flag."
-        )
-
+    flag_inputs = {
```

Comment:
Why create a dict and not just directly pass the arguments to `_validate_flag_inputs`?

Reply:
I think it's in general a good idea - to put all CLI inputs to one dictionary, because currently there are a lot of them, but it should be done not only for one _validate_flag_inputs() function, but to the whole subsequent code

Reply:
I agree, the initial reason was due to how many inputs there are, and for future implementation/refactor as there is no guarantee that we won't need to pass these through later. I couldn't find any strong support for either convention, however, so I am mostly ambivalent. Though whatever we do choose, we should be sure to implement consistently.